playbooks/roles/streisand-gateway/templates/openssl-local.cnf.j2

```diff
@@ -0,0 +1,85 @@
+distinguished_name = req_distinguished_name
+
+[ ca ]
+default_ca = CA_default
+
+[ CA_default ]
+
+dir = {{ openssl_ca_base }}
+certs = $dir
+crl_dir = $dir
+database = $dir/index.txt
+new_certs_dir = {{ openssl_ca_newcerts_dir }}
+
+certificate = {{ openssl_ca_certificate }}
+serial = $dir/serial
+crl = $dir/crl.pem
+private_key = {{ openssl_ca_key }}
+RANDFILE = $dir/.rand
+
+copy_extensions = copy
+
+x509_extensions = v3_ca
+
+default_days = {{ nginx_days_valid }}
+default_crl_days= 30
+default_md = {{ openssl_ca_default_md }}
+preserve = no
+
+policy = policy_anything_including_no_cn
+
+keyUsage = cRLSign, keyCertSign
```

Comment:
I don't think `keyUsage` does anything here. That's an X509v3 attribute and I think it probably gets ignored here entirely (or ends up being stomped by the `x509_extensions` configuration in `v3_ca` when it sets `keyUsage = cRLSign, dataEncipherment, digitalSignature, keyCertSign, keyEncipherment, nonRepudiation`). I don't see it in the phildev `CA_default` example either.

Reply:
The place I was looking was http://openssl.cs.utah.edu/docs/apps/x509.html#certificate_extensions (well, the local version) and yeah, most of them have the "EKU must be absent, or...", so that suggests nuking the EKUs.

Does the list of tests in the manual have anything to do with the real world these days? Netscape doesn't ship a web s...wait, there isn't a Netscape.

Reply:
I nuked all the KU and EKU.